src/App.tsx

```diff
@@ -17,18 +37,33 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                filter={filter}
+                setFilter={setFilter}
+                query={query}
+                setQuery={setQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading && (<Loader />)}
+              <TodoList
+                filteredTodos={filteredTodos}
+                setSelectedTodoId={setSelectedTodoId}
+                selectedTodo={selectedTodo}
+              />
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo && (
+        <TodoModal
+          selectedTodoId={selectedTodoId}
+          selectedTodo={selectedTodo}
```

Comment:
Why do you need to pass both of these props, can you get `id` from `selectedTodo` object?